libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/RestResponse.java

```diff
@@ -189,6 +190,24 @@ public JSONObject asJSONObject() throws JSONException, IOException {
 		return responseAsJSONObject;
 	}
 
+	/**
+	 * JSONObject including the response body, status code and headers.
+	 *
+	 * @return JSONObject for full response
+	 * @throws JSONException
+	 * @throws IOException
+	 */
+	public JSONObject fullResponseAsJSONObject() throws JSONException, IOException {
```

Comment:
This extra method is a bit weird. The method `asJsonObject()` is meant to be a JSON representation of this object. I'd rather we use that method. The receiver can extract all the fields from that or parse it however they want.

Reply:
Could we revert changes to this file and go with `asJsonObject()` instead?

Reply:
`asJsonObject` just has the response body but not status code & headers. Do you want me to move this part to a helper method in the bridge and plugin classes?

Reply:
Do we need `status code` and `headers` in the error callback? If we really need it, then a helper method in the bridge and plugin classes would be less confusing IMO.

Reply:
The code and headers will probably have more value in the case of failed requests. So this method will be more valuable than the existing one.